### PR TITLE
Add sigs.k8s.io to gitignore

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

this folder is generated by `make generate` command and should not be included into git file

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -170,3 +170,6 @@ zz_generated.openapi.go
 
 # sshuttle
 sshuttle.pid
+
+# manifests
```
Comment (sbueringer) [inline .gitignore:174]: That's an error in the generation nothing which we should add to gitignore..

Comment (sbueringer) [reply .gitignore:174]: I  think it happens when the repo is in the GOPATH but not under $GOPATH/src/sigs.k8s.io/cluster-api-provider-openstack

xref: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/801

Comment (sbueringer) [reply .gitignore:174]: @jichenjc I assumed this happened with the current master.

How is your setup regarding GOPATH and folder location?

Comment (jichenjc) [reply .gitignore:174]: yes, master, 
GOPATH is not set, I only run `make generate` at my folder 

~/go/src/sigs.k8s.io/cluster-api-provider-openstack/

previously there is no such issue and after some changes recently I noticed such file exist

Comment (sbueringer) [reply .gitignore:174]: Okay thx. I"ll try to fix it

Comment (sbueringer) [reply .gitignore:174]: This might fix it: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/827

**Pull request closed without merging**